src/Authorization/Services/Implementation/ContextHandler.cs

```diff
@@ -103,7 +99,13 @@ protected async Task EnrichResourceAttributes(XacmlContextRequest request, bool
                 else
                 {
                     instanceData = new();
-                    AuthInfo authInfo = await _policyInformationRepository.GetAuthInfo(resourceAttributes.InstanceValue);
+                    
+                    if (!authInfoCache.TryGetValue(resourceAttributes.InstanceValue, out AuthInfo authInfo))
```

Comment:
This is most likely not thread safe

Reply:
The cache is an instance variable from the controller. It's only used for evaluations for one decision request. The processing within a decision request is always sequential.